lib/logstash/java_integration.rb

```diff
@@ -24,24 +24,39 @@ def self.===(other)
   end
 end
 
+# map_mixin to patch LinkedHashMap and HashMap. it must be done directly on the classes,
+# using a module mixin does not work, and injecting in the Map interface does not work either
+# but injecting in the class works.
 
-# this is a temporary fix to solve a bug in JRuby where classes implementing the Map interface, like LinkedHashMap
-# have a bug in the has_key? method that is implemented in the Enumerable module that is somehow mixed in the Map interface.
-# this bug makes has_key? (and all its aliases) return false for a key that has a nil value.
-# Only LinkedHashMap is patched here because patching the Map interface is not working.
-# TODO find proper fix, and submit upstream
-# releavant JRuby files:
-# https://github.com/jruby/jruby/blob/master/core/src/main/ruby/jruby/java/java_ext/java.util.rb
-# https://github.com/jruby/jruby/blob/master/core/src/main/java/org/jruby/java/proxies/MapJavaProxy.java
-class Java::JavaUtil::LinkedHashMap
+map_mixin = lambda do
```

Comment:
well, good point, most/all of the java_integration stuff could be considered a patch... can we merge this and create a ticket to see if we should refactor into lib/logstash/patches?